src/com/itschool/session4/homework/ex1/AssigmentCompound2.java

```diff
@@ -0,0 +1,23 @@
+package com.itschool.session4.homework.ex1;
+import java.util.Scanner;
+public class AssigmentCompound2 {
+    public static void main(String[] args) {
+
+        Scanner sc = new Scanner(System.in);
+
+        System.out.println("Enter the Original price: ");
+        double ogPrice = sc.nextDouble();
+
+        System.out.println("Enter the discount percentage: ");
+        double discountPercentage = sc.nextDouble();
+
+        double discountAmount = ogPrice * (discountPercentage / 100);
+
+        ogPrice -= discountAmount;
+
+        System.out.printf("The price after a %.2f%% discount is: %.2f" , discountPercentage, ogPrice);
+
+        sc.close();
```

Comment:
I'm closing the scanner object and cannot be used anymore, should i keep scanner open?

Reply:
Closing resources is the right way (most of the times also the only way) to the things working correctly, but in this case is not mandatory since the scanner wraps the builtin jdk input system. Moreover your program exits after finishing the code from this class, therefore all resources are being destroyed.
In short, in this situation does not make any difference. I just want to check why you wrote this line :p